src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/CancelFlow.tsx

```diff
@@ -4,42 +4,182 @@
+      setCouponSuccess(true);
+    } else {
+      setCouponSuccess(false);
+    }
+  };
+
+  const checkCouponCode = async () => {
+    const result = await onCheckUserHasCurrentCouponSet();
+    if (typeof result.success === "boolean") {
+      setAlreadyHasCouponSet(result.success);
+    } else {
+      setAlreadyHasCouponSet(false);
+    }
+  };
+
+  useEffect(() => {
+    void checkCouponCode();
+    // Only called once upon initial render to check if a user had a coupon code previously set
+  }, []);
+
+  useEffect(() => {
+    if (couponSuccess) {
+      setCurrentStep("all-set");
+    }
+  }, [couponSuccess]);
+  /* c8 ignore stop */
+
   const dialogTrigger = useOverlayTrigger({ type: "dialog" }, dialogState);
 
+  const dialogTitle = () => {
+    switch (step) {
+      case "confirm":
+        return "settings-cancel-plus-step-confirm-heading";
+      case "all-set":
+        return "settings-unsubscribe-dialog-promotion-complete";
+      case "redirecting":
+        return "settings-unsubscribe-dialog-confirmation-redirect-title";
+      case "survey":
+        return "settings-cancel-plus-step-survey-heading";
+    }
+  };
+
+  const Animation = () => {
+    return (
+      <>
+        <video
+          aria-hidden={true}
+          autoPlay={true}
+          loop={true}
+          muted={true}
+          className={styles.cancellationAnimation}
+        >
+          <source
+            // Unfortunately video files cannot currently be imported, so make
+            // sure these files are present in /public. See
+            // https://github.com/vercel/next.js/issues/35248
+            type="video/mp4"
+            src={
+              step === "all-set"
+                ? /* c8 ignore next */
+                  "/animations/CancellationFlowDiscountAppliedAnimation.mp4"
+                : "/animations/CancellationFlowAnimation.mp4"
+            }
+          />
+          <source
+            type="video/webm"
+            src={
+              step === "all-set"
+                ? /* c8 ignore next */
+                  "/animations/CancellationFlowDiscountAppliedAnimation.webm"
+                : "/animations/CancellationFlowAnimation.webm"
+            }
+          />
+          {/* Fall back to the image if the video formats are not supported: */}
+          <Image
+            className={styles.cancellationIllustrationWrapper}
+            src={
+              /* c8 ignore next */
+              step === "all-set"
+                ? CancellationFlowDiscountAppliedStaticImage
+                : CancellationFlowStaticImage
+            }
+            alt=""
+          />
+        </video>
+        {/* Fall back to the image if the video formats are not supported: */}
+        {/* The .staticAlternative class ensures that this image will only be shown if the user has prefers-reduced-motion on */}
+        <Image
+          className={`
+${styles.cancellationIllustrationWrapper}
+${styles.staticAlternative}
+`}
```

Comment:
Formatting: Not sure why the linter is not picking up on this, but the formatting seems to be off here?